File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int idx1 = 0;
        int idx2 = 0;
        
        for(int i=0; i<goal.length; i++) {
            String g = goal[i];
            if(idx1 < cards1.length && g.equals(cards1[idx1])) {
                idx1++;
                continue;
            }
            
            if(idx2 < cards2.length && g.equals(cards2[idx2])) {
                idx2++;
                continue;
            }
            
            return "No";
        }
        return "Yes";
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    var answer = '';
    for(const i of goal){
        if(i===cards1[0]){
            cards1.shift()
        } else if(i===cards2[0]){
            cards2.shift()
        } else{
            return 'No'
        }
    }
    
    return 'Yes';
}